### PR TITLE
Initial PR for L9 Support

### PR DESCRIPTION
**Pull request by sniper7kills**
Initial PR in order to support Laravel 9.

Major change is the dropping of the Swift Mailer in favor of the Symphony Mailer and thus the change of:
`replaceSwiftMailer` for `replaceSymfonyTransport`

### CODE REVIEW

**Comment by fletch3555** [comment]
Jobs are failing due to https://github.com/tenancy/tenancy/runs/5161472457?check_suite_focus=true#step:4:351

All the required jobs are showing up because you removed all the older laravel/php versions from the test job, but they're marked as required in the repository settings.

**Comment by Grummfy** [comment]
any update on this PR?

**Comment by sniper7kills** [comment]
@Grummfy Unfortunately I do not have any update; I know that @ArlonAntonius has been extremally busy recently.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,35 +15,12 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: ['7.2', '7.3', '7.4', '8.0']
-                laravel: [7.*, 8.*]
+                php: ['8.0']
```
Comment (ArlonAntonius) [inline .github/workflows/test.yml:18]: Should support PHP 8.1 as well :)

File: src/Affects/Mails/Events/ConfigureMails.php [hunk 2]
```diff
@@ -46,9 +45,10 @@ public function __construct(Switched $event, Mailer $mailer)
      *
      * @return self
      */
-    public function replaceSwiftMailer(Swift_Transport $transport)
+    public function replaceSymfonyTransport(TransportInterface $transport)
```
Comment (ArlonAntonius) [inline src/Affects/Mails/Events/ConfigureMails.php:48]: just replaceTransport would be enough for me

File: tests/Affects/Mails/Unit/ConfigureMailsTest.php [hunk 3]
```diff
@@ -30,14 +30,14 @@ class ConfigureMailsTest extends AffectsEventUnitTestCase
     protected $event = ConfigureMails::class;
 
     /** @test */
-    public function the_event_can_replace_swiftmailer()
+    public function the_event_can_replace_symfony_transport()
```
Comment (ArlonAntonius) [inline tests/Affects/Mails/Unit/ConfigureMailsTest.php:33]: Idem

File: phpunit.xml [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
+    <env name="SESSION_DRIVER" value="array"/>
+    <env name="QUEUE_DRIVER" value="sync"/>
+    <env name="DB_CONNECTION" value="sqlite"/>
+    <env name="DB_DATABASE" value=":memory:"/>
+    <env name="TENANCY_MYSQL_HOST" value="mysql"/>
+    <env name="TENANCY_DB" value="testing"/>
+    <env name="TENANCY_USERNAME" value="mysql_admin"/>
+    <env name="TENANCY_PASSWORD" value="mysql_admin_password"/>
+  </php>
 </phpunit>
```
Comment (ArlonAntonius) [inline phpunit.xml:40]: What is this? Linting? Update or something else?

**Review by ArlonAntonius — CHANGES_REQUESTED** [review]
Do we have PHP8.0 and PHP8.1 images released in order to support these in the CI?

File: phpunit.xml (hunk 4 above)
Comment (Tofandel) [reply phpunit.xml:40]: It's the new config format for phpunit 9 (it was upgraded) and If I remember correctly it upgrades the config automatically when running the tests

**Comment by Tofandel** [comment]
Btw shouldn't it go into a new 2.x branch? Or maybe let's keep BC by just adding the replaceTransport method instead of renaming the old one, adding the php matrix for laravel 9 and skipping test as needed by laravel version

Because I don't think just one method change justifies a major version

**Comment by torstenlemke** [comment]
any updates?

**Comment by mennovanhout** [comment]
> any updates?

https://github.com/tenancy/tenancy/pull/250

**Comment by madalinignisca** [comment]
Ush. Let's add also php 8.1 and get this 2.x rolling in.

File: src/Affects/Mails/Events/ConfigureMails.php (hunk 2 above)
Comment (bretto36) [reply src/Affects/Mails/Events/ConfigureMails.php:48]: The Laravel 9 Methods all seem to say Symfony in them so perhaps we should leave it as is?

`$this->mailer->setSymfonyTransport($transport);`

```
$this->withSymfonyMessage(function (\Symfony\Component\Mime\Email $message) {
$message->getHeaders()->addTextHeader('TEST', 'TEST');
}
```

**Comment by bretto36** [comment]
Any Update on this? or the other PR?

**Comment by ArlonAntonius** [comment]
Superseeded by #252
Which will get merged soon

(CC: @bretto36 @madalinignisca @torstenhoegel @Tofandel )

**Pull request closed without merging**